### ISSUE TITLE
UI/UX: control panel - gripes about Overlays panel

### ISSUE THREAD

**Issue by TaxiService** [DFHack/dfhack#2774]
the Overlays section of the DFHack Control Panel acts a bit weird.

these steps produce what's shown in the following screenshot:
- open control panel and switch to Overlays panel.
- click on `[configure]` next to `unsuspend.overlay`, doing so opens the `Overlay config` window
- in it, disable `hotkeys.menu` by clicking on `[enable]`

![Capture](https://user-images.githubusercontent.com/5853667/216056619-2844df99-b9d5-4d0f-85af-6c6c62b7c38c.PNG)

here, two things worked unexpectedly:
- in step 2, i would expect `[configure]` to open _unsuspend.overlay_'s configuration window; not another panel that lets me disable all tools individually just like the previous panel.
- in step 3, when disabling `hotkeys.menu` from the config window, the activation-button in controlpanel's window is unaffected. (until the panel is somehow "refreshed", like by switching to another panel and back, or closing/reopening the control panel)

so in short:
- having a `[configure]` button next to each individual tool feels redundant, since they all open the same window
- having `[enabled]`/`[disabled]` buttons in the Overlay config window _also_ feels redundant, since i can already disable them from the previous window
- not having the tool's status update in the panel after toggling it in the config window, feels confusing.

**Comment by myk002** [comment]
Apart from technical difficulties in dealing with painting full-screen from a widget embedded several layers deep in the hierarchy, one thing that kept me from wholly moving `gui/overlay` into `gui/control-panel` is the different Screen interface modes. `gui/control-panel` is a "regular" ZScreen that you can defocus and park behind other windows. `gui/overlay` is a basic Screen (not a ZScreen) that halts all processing of the lower layers since it is interacting with screen elements. For example, the logo hotspot functions when the control panel is shown, but it does not function when gui/overlay has control (by design -- otherwise you wouldn't be able to move it around). We could make the gui/control-panel ZScreen modal while the overlays page is showing, but I'm worried that would be  a jarring user experience. No other tool behaves like that. I'm definitely open to ideas. This is a thornier problem than I thought at first.

**Comment by myk002** [comment]
> i removed the `(no repositionable panel)` from unsuspend because... in this case the user does not really need to know that, right?

That's there to explain why there is no highlighted panel when that list item is selected. Other overlays have widgets to reposition that are highlighted when their list item is highlighted. The player would expect there to be a corresponding highlight for `unsuspend` as well, and could be confused when they couldn't find it.

**Comment by TaxiService** [comment]
yeah, i bet that it's a significant pain in the urethra to solve, code-wise!

another way this could be "solved" would be having a mix of the two mockups i made.
basically imagine the second mockup, but with a `[reposition panels]` button like in the first mockup.

when clicking that button, the new "overlay config" window would be a small panel that appears in the middle of the screen, possibly with instructions for how to reposition the panels.
This window could be the "basic Screen" type, and allow you to edit the positions of panels.
This window should _only_ let you reposition things around, and not let you enable/disable anything.

in this way, the enable/disable controls and the help buttons only appear in the control panel, _and_ you'd get the benefit of being able to not deal with those problems you mentioned. maybe?

**Comment by TaxiService** [comment]
> That's there to explain why there is no highlighted panel when that list item is selected. Other overlays have widgets to reposition that are highlighted when their list item is highlighted. The player would expect there to be a corresponding highlight for `unsuspend` as well, and could be confused when they couldn't find it.

would they, though? I mean, following my other comment, if they did click the `[reposition panels]` button... they would be shown which panels can be repositioned.
if unsuspend shows nothing, then it means that it has no panels to reposition! i believe it isn't strictly necessary to tell the user that that particular tool has no panels to reposition. : |

(my problem with it is that that message has a chance to not fit, if a tool's name is longer or whatever. and idk, if the user read what unsuspend does, they'd not expect panels to move or anything)

**Comment by TaxiService** [comment]
here, something like this.

![mock3](https://user-images.githubusercontent.com/5853667/216109216-3042d02c-2c5c-440e-9219-bfd3cfda9171.png)

so that the "Overlay config" window only lets you reposition the panels, and enable/disable + help are available only from the control panel.

you could probably change the "Overlay config" window name to something like "Reposition panels" or the like

**Comment by TaxiService** [comment]
amazing!!! <3<3<3
